packages/compass-e2e-tests/index.ts

```diff
@@ -1,22 +1,25 @@
-const path = require('path');
-const fs = require('fs');
-const { promisify } = require('util');
-const glob = require('glob');
-const { sync: spawnSync } = require('cross-spawn');
-const Mocha = require('mocha');
-const debug = require('debug')('compass-e2e-tests');
-const {
+#!/usr/bin/env ts-node
+import path from 'path';
+import fs from 'fs';
+import { promisify } from 'util';
+import glob from 'glob';
+import crossSpawn from 'cross-spawn';
+import Mocha from 'mocha';
+import Debug from 'debug';
+import {
   rebuildNativeModules,
   compileCompassAssets,
   buildCompass,
   LOG_PATH,
-} = require('./helpers/compass');
-const { createUnlockedKeychain } = require('./helpers/keychain');
-const ResultLogger = require('./helpers/result-logger');
+} from './helpers/compass';
+import { createUnlockedKeychain } from './helpers/keychain';
+import ResultLogger from './helpers/result-logger';
 
+const debug = Debug('compass-e2e-tests');
 const keychain = createUnlockedKeychain();
 
-let metricsClient;
+// We can't import mongodb here yet because native modules will be recompiled
+let metricsClient: any;
```

Comment:
You should still be able to do something like this if you want to:

```suggestion
import type { MongoClient } from 'mongodb';
let metricsClient: MongoClient;
```

Reply:
The problem there was with importing mongodb before it gets rebuilt. Caused really obscure failures that took me days to find.

Reply:
Right, but `import type` really is a type-only import, i.e. would not be affected by rebuilds at all because it doesn’t actually load the module at runtime